app/autotranslate/server/settings.js

```diff
@@ -3,6 +3,37 @@ import { Meteor } from 'meteor/meteor';
 import { settings } from '../../settings';
 
 Meteor.startup(function() {
-	settings.add('AutoTranslate_Enabled', false, { type: 'boolean', group: 'Message', section: 'AutoTranslate', public: true });
-	settings.add('AutoTranslate_GoogleAPIKey', '', { type: 'string', group: 'Message', section: 'AutoTranslate', enableQuery: { _id: 'AutoTranslate_Enabled', value: true }, secret: true });
+	settings.add('AutoTranslate_Enabled', false, {
+		type: 'boolean',
+		group: 'Message',
+		section: 'AutoTranslate',
+		public: true,
+	});
+
+	settings.add('AutoTranslate_ServiceProvider', 'google-translate', {
+		type: 'select',
+		group: 'Message',
+		section: 'AutoTranslate',
+		values: [{
+			key: 'google-translate',
+			i18nLabel: 'AutoTranslate_Google',
+		}, {
+			key: 'deepl-translate',
+			i18nLabel: 'AutoTranslate_DeepL',
+		}],
+		enableQuery: [{ _id: 'AutoTranslate_Enabled', value: true }],
+		i18nLabel: 'AutoTranslate_ServiceProvider',
+		public: true,
+	});
+
+	settings.add('AutoTranslate_APIKey', '', {
```

Comment:
any special reason to have only one setting as the API Key instead of one for each service? I would expect something like we have for File Storage, where each service provider has its own settings (which includes api key and secret but not only that).

Reply:
Not really. Originally, It was designed such a way to keep the settings section generic and compact (only with endpoint and secret) and as it is obvious that only one provider can be active at a time.  I have adapted now the design as per you suggestion. However, auto-translate is sub section under message group and having own settings for each provider will lead to additional sub sections among other sections under Messages group which might additional complexity imho.